test/Semantics/resolve86.f90

```diff
@@ -0,0 +1,21 @@
+! RUN: %B/test/Semantics/test_errors.sh %s %flang %t
+! C736 If EXTENDS appears and the type being defined has a coarray ultimate 
+! component, its parent type shall have a coarray ultimate component.
+!
+subroutine s()
+  type coarrayParent
+    real, codimension[*] :: parentField
+  end type coarrayParent
+
+  type, extends(coarrayParent) :: goodChildType
+    real, codimension[*] :: childField
+  end type goodChildType
+
+  type plainParent
+  end type plainParent
+
+  !ERROR: Type 'badchildtype' has a coarray ultimate component so the type it extends ('plainparent') must have a coarray ultimate component
+  type, extends(plainParent) :: badChildType
+    real, codimension[*] :: childField
+  end type badChildType
+end subroutine s
```

Comment:
It wouldn't hurt to have a test where the type has a coarray ultimate component that is not a coarray component (i.e. that verifies it still does the right thing in the "ultimate" case).

Reply:
I don't understand.  The test includes the type `goodChildType` that has a coarray ultimate component and extends `coarrayParent` that also has a coarray ultimate component.  I'm not sure what you mean by "a coarray ultimate component that is not a coarray component".

Reply:
I was thinking something like:
```
type, extends(plainParent) :: t1
  type(t2) :: x
end type
```
`t1` doesn't have a coarray component but this should fail if `t2` has a coarray ultimate component.

Reply:
Got it.  I'll add a test.